### PR TITLE
AAC Fixes

### LINKED ISSUE #13287: Leaving a long AAC script running causes an ungodly amount of lag
Recently, an AAC script that disabled all (or most) of the pressure checks on station began seeing some use.

While you can set it to run and disable it immediately after to disable all the pressure checks in one swoop, leaving it running will I guess constantly attempt to disable all pressure checks. This causes an immense amount of stuttering for almost everyone involved that makes it an absolute pain in the ass to play with.

### PR DESCRIPTION
**Pull request by N3X15**
# Executive Summary
Fixes and improvements for the Atmospherics Automation Console.

Fixes #13287, #12920

# Testing
dreammaker_2017-02-12_20-35-47

1. Modified tiny to add a simple air mixing setup using two injectors, a vent, an O2 and N2 can, and the air mixing AAC preset.
2. Spawned as MoMMI and verified stop/start works.
3. Verified clicking on run once produces "Execution Complete" in chat.
4. Verified clicking again results in a "can only run once every 5 seconds" warning.
5. F5'd to admin ghost
6. Verified stop/start worked.
7. Verified I could change settings in the script
8. De-admin'd.
9. Verified I could NOT toggle stop/start.
10. Verified I could NOT change script settings.

# Changelog
:cl:
* rscadd: "AAC now has a \\"Run Once\\" button, which can only be used once every 5 seconds."

### CODE REVIEW

File: code/modules/atmos_automation/statements.dm [hunk 1]
```diff
@@ -105,31 +105,46 @@ var/global/automation_types=typesof(/datum/automation) - /datum/automation
...
+		var/ghost_flags = 0
+		if(parent.ghost_write)
+			ghost_flags |= PERMIT_ALL
+		if(canGhostWrite(O,parent,"",ghost_flags) || isAdminGhost(O))
+			if(!parent.custom_aghost_alerts)
+				log_adminghost("[key_name(user)] screwed with [parent] ([href])!")
+			return TRUE // Ghost is admin or otherwise allowed to dick with things.
+		return FALSE // Ghost is NOT admin? Bail.
+
+	if(user.restrained() || user.lying || user.stat)
```
Comment (PJB3005) [inline code/modules/atmos_automation/statements.dm]: `user.incapacitated()` gets the `user.restrained() || user.stat` checks here. Doesn't seem like a helper for lying exists though.

File: code/modules/atmos_automation/statements.dm [hunk 2]
```diff
@@ -105,31 +105,46 @@ var/global/automation_types=typesof(/datum/automation) - /datum/automation
...
+		to_chat(user, "<span class='warning'>You don't have the dexterity to do this!</span>")
+		return FALSE
+
+	// Silicons can access shit from out of range.
+	if(istype(user, /mob/living/silicon))
+		return TRUE
+
+	// RAAAANGE CHEEEECK
+	var/norange = 0
+	if(user.mutations && user.mutations.len)
```
Comment (PJB3005) [inline code/modules/atmos_automation/statements.dm]: `in_range` already handles TK it seems.

File: code/modules/atmos_automation/statements.dm [hunk 3]
```diff
@@ -105,31 +105,46 @@ var/global/automation_types=typesof(/datum/automation) - /datum/automation
...
+		if(M_TK in user.mutations)
+			norange = 1
+
+	if(!norange)
+		if ((!in_range(parent, user) || !istype(parent.loc, /turf)))
```
Comment (PJB3005) [inline code/modules/atmos_automation/statements.dm]: `isturf`

**Review by PJB3005 — CHANGES_REQUESTED** [review]

**Comment by N3X15** [comment]
Done @Probe1

**Review by PJB3005 — APPROVED** [review]

**Pull request merged**